py/emitnative.c

```diff
@@ -1873,27 +1868,28 @@ static void emit_native_store_subscr(emit_t *emit) {
                 }
                 case VTYPE_PTR32: {
                     // pointer to 32-bit memory
+                    #if N_THUMB
+                    if (index_value >= 0 && index_value < 32) {
+                        asm_thumb_str_rlo_rlo_i5(emit->as, reg_value, reg_base, index_value);
+                        break;
+                    }
+                    #elif N_RV32
+                    if (FIT_SIGNED(index_value, 10)) {
+                        asm_rv32_opcode_sw(emit->as, reg_value, reg_base, index_value << 2);
+                        break;
+                    }
+                    #elif N_XTENSA || N_XTENSAWIN
+                    if (index_value >= 0 && index_value < 256) {
+                        asm_xtensa_s32i_optimised(emit->as, reg_value, reg_base, index_value);
```

Comment:
I see you've fixed a subtle bug here: `reg_value` used to be `REG_RET`!

Nice one.